Review

kspshnik opened the pull request.

kspshnik committed:
[Refactor] Set `viewBox` in `floppy.svg`

kspshnik committed:
[Fix] Corrected quotation marks for one of `url()`'s

kspshnik committed:
[New] Added styles for  `.page` & `.footer`

kspshnik committed:
[Feature] Added full-fledge header

kspshnik committed:
[Feature] Implemented basic button animations

kspshnik committed:
[Final] Project is ready for review.

kspshnik committed:
[Fix] Punctuation in README.md

kspshnik committed:
[Fix] Partially fixed code layout

AliaksandrHrychonak commented on index.html:
```diff
@@ -31,7 +31,9 @@
 </head>
```
Отлично

head оформлен верно

AliaksandrHrychonak commented on index.html:
```diff
@@ -31,7 +31,9 @@
-  <header class="header">
+  <header class="header content-section bordered-section">
+    <p class="header__logo">&lt;/HTML&gt;</p>
+    <h1 class="header__title">закрывающий тэг</h1>
```
Отлично

верный уровень заголовка

AliaksandrHrychonak commented on index.html:
```diff
@@ -53,7 +55,70 @@
+  <main class="content content-section bordered-section">
+    <ul class="stories">
+      <li class="stories__item">
+        <article class="card content-section">
```
Отлично

Верное использование тега article

AliaksandrHrychonak commented on index.html:
```diff
@@ -53,7 +55,70 @@
+            материал :)
+          </p>
+          <footer class="card__buttons">
+            <button aria-labelledby="like-button" class="card__icon-button">
```
надо исправить

  Кнопкам на странице необходимо добавить атрибут `type` с соответствующим значением. Это важно для правильного функционирования и семантики HTML. Для большинства кнопок подойдет значение `type="button"`, если они не отправляют форму. Для кнопок, отправляющих форму, используйте `type="submit"`.

Добавление атрибута `type` улучшит доступность вашего сайта и сделает его более совместимым с различными браузерами и устройствами.

kspshnik replied:
Fixed

AliaksandrHrychonak commented on index.html:
```diff
@@ -53,7 +55,70 @@
+                  fill="var(--heart-color)" />
+              </svg>
+            </button>
+            <button id="like-button" class="button card__like-button">
```
надо исправить

Кнопкам на странице необходимо добавить атрибут `type` с соответствующим значением. Это важно для правильного функционирования и семантики HTML. Для большинства кнопок подойдет значение `type="button"`, если они не отправляют форму. Для кнопок, отправляющих форму, используйте `type="submit"`.

Добавление атрибута `type` улучшит доступность вашего сайта и сделает его более совместимым с различными браузерами и устройствами.

kspshnik replied:
Fixed.

AliaksandrHrychonak commented on index.html:
```diff
@@ -53,7 +55,70 @@
+          </p>
+          <footer class="card__buttons">
+            <button aria-labelledby="like-button" class="card__icon-button">
+              <svg class="like-icon" width="39" height="36" viewBox="0 0 39 36" fill="none"
```
можно лучше 

Данный элемент не несёт смысловой нагрузки, было бы хорошим решением добавить `aria-hidden="true"`.

Подробнее => https://doka.guide/a11y/aria-hidden/

kspshnik replied:
Fixed.

AliaksandrHrychonak commented on index.html:
```diff
@@ -53,7 +55,70 @@
+    </ul>
+
+  </main>
+  <footer class="footer">
```
надо исправить

Кнопка относится к основному содержимому страницы, разместите ее внутри `main` тега, а обёртку тегом `<footer>` следует удалить.

Использование тега `footer` для обертки кнопки "Сохранить на память" не является корректным в данном контексте. Тег `footer` предназначен для разметки подвала страницы или секции, содержащего информацию о авторских правах, контактные данные или ссылки на связанные документы. Одиночная кнопка открытия модального окна не соответствует этому назначению.

kspshnik replied:
Fixed.

AliaksandrHrychonak commented on styles/animations.css:
```diff
@@ -1,14 +1,105 @@
+
+.like-icon .like-icon__core {
+  fill: none;
+  transition: fill 0.3s linear;
```
надо исправить

по заданию при переходе есть задержка

kspshnik replied:
Fixed

AliaksandrHrychonak commented on styles/animations.css:
```diff
@@ -1,14 +1,105 @@
+
+.like-icon:hover .like-icon__core {
+  fill: var(--accent-color);
+  transition: fill 0.3s 0.03s linear;
```
надо исправить

по заданию при переходе нет задержки

kspshnik replied:
Fixed

AliaksandrHrychonak commented on styles/animations.css:
```diff
@@ -1,14 +1,105 @@
+
+.like-icon .like-icon__main-body {
+  fill: none;
+  transition: fill 0.05s 0.3s linear;
```
надо исправить

время перехода и задержка не по заданию

kspshnik replied:
fixed

AliaksandrHrychonak commented on styles/animations.css:
```diff
@@ -1,14 +1,105 @@
+
+.like-icon:hover .like-icon__main-body {
+  fill: var(--accent-color);
+  transition: fill 0.3s linear;
```
надо исправить

по заданию при переходе есть задержка

kspshnik replied:
Fixed

AliaksandrHrychonak commented on styles/animations.css:
```diff
@@ -1,14 +1,105 @@
+
+.like-icon:active .like-icon__core {
+  fill: var(--heart-color);
+  transition: fill 0.3s 0.03s linear;
```
надо исправить

по заданию при переходе нет задержки

kspshnik replied:
Fixed

AliaksandrHrychonak commented on styles/animations.css:
```diff
@@ -1,14 +1,105 @@
+
+.like-icon.is-liked .like-icon__core {
+  fill: var(--heart-color);
+  transition: fill 0.3s 0.03s linear;
```
надо исправить

по заданию при переходе нет задержки

kspshnik replied:
Fixed

AliaksandrHrychonak commented on styles/animations.css:
```diff
@@ -1,14 +1,105 @@
+
+.like-icon:active .like-icon__main-body {
+  fill: var(--heart-color);
+  transition: fill 0.3s linear;
```
надо исправить

по заданию при переходе есть задержка

kspshnik replied:
Fixed

AliaksandrHrychonak commented on styles/animations.css:
```diff
@@ -1,14 +1,105 @@
+
+.like-icon.is-liked .like-icon__main-body {
+  fill: var(--heart-color);
+  transition: fill 0.3s linear;
```
надо исправить

по заданию при переходе есть задержка

kspshnik replied:
Fixed

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -21,12 +21,350 @@
+  font-weight: 400;
+  background-image: var(--background-pattern);
+  background-repeat: repeat;
+  background-attachment: fixed;
```
отлично 

фон фиксирован

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -21,12 +21,350 @@
+{
+  outline: none;
+  box-shadow: 2px 2px 0 0 var(--accent-color);
+  transition: box-shadow 0.3s ease;
```
надо исправить

Правило `transition` следует добавить один раз к основному селектору. Так эффект будет работать плавно в обе стороны - когда стиль добавляется и когда убирается.

kspshnik replied:
Fixed

AliaksandrHrychonak reviewed (commented):
Необходимо развернуть общий комментарий ⬇️

Добрый день. Работа выполнена на хорошем уровне! 

Отлично:

* Размечена вся структура
* Используете семантичные теги для разметки.
* аккуратно оформленный код
* все комментарии "Отлично"

Надо исправить:

* Все комментарии с пометкой "надо исправить"
* Все Error валидатора https://validator.w3.org/#validate_by_input
* Всем кнопкам в проекте нужен атрибут `type`
* Иконка лайка `like-icon` https://disk.yandex.com/i/VnAfyo-e7rU4YQ
  * Иконку лайка необходимо центрировать в кнопке(удобно использовать сетку `flex/grid`)
  * Высота рамки при фокусе на иконку лайка согласно заданию сходится с высотой кнопки `Like`
  * Между кнопкой с иконкой лайка и кнопкой `Like` согласно заданию есть разрыв
* В проекте есть нарушения в отображении по макету
    * **На 1440px не по макету:**
      * https://disk.yandex.com/i/MKnW7yM2YcqAAA
    * **На 375px не по макету:**
      * https://disk.yandex.com/i/J-TwxrvC881TEQ
    * Для более точного контроля отображения вёрстки рекомендуется использовать плагин `PerfectPixel` для браузера `Chrome`. Этот инструмент позволяет накладывать изображение макета поверх вёрстки в браузере, что значительно упрощает процесс выявления и исправления визуальных несоответствий. Для этого достаточно экспортировать весь макет из Figma в формате изображения и использовать его в качестве overlay-слоя при помощи [плагина](https://chromewebstore.google.com/detail/perfectpixel-by-welldonec/dkaagdgjmgdmbnecmcefdhjekcoceebi){target="_blank"}


Можно лучше:

* Все комментарии "Можно лучше" в коде.
* В качестве альтернативного шрифта для `PressStart2P` по типу засечек можно использовать `fantasy`

Необходимо исправить все "надо исправить". Исправление "можно лучше" остаются на ваше усмотрение.

AliaksandrHrychonak commented:
> Необходимо развернуть общий комментарий ⬇️
> 
> Добрый день. Работа выполнена на хорошем уровне!
> 
> Отлично:
> 
> * Размечена вся структура
> * Используете семантичные теги для разметки.
> * аккуратно оформленный код
> * все комментарии "Отлично"
> 
> Надо исправить:
> 
> * Все комментарии с пометкой "надо исправить"
> * Все Error валидатора https://validator.w3.org/#validate_by_input
> * Всем кнопкам в проекте нужен атрибут `type`
> * Иконка лайка `like-icon` https://disk.yandex.com/i/VnAfyo-e7rU4YQ
>   
>   * Иконку лайка необходимо центрировать в кнопке(удобно использовать сетку `flex/grid`)
>   * Высота рамки при фокусе на иконку лайка согласно заданию сходится с высотой кнопки `Like`
>   * Между кнопкой с иконкой лайка и кнопкой `Like` согласно заданию есть разрыв
> * В проекте есть нарушения в отображении по макету
>   
>   * **На 1440px не по макету:**
>     
>     * https://disk.yandex.com/i/MKnW7yM2YcqAAA
>   * **На 375px не по макету:**
>     
>     * https://disk.yandex.com/i/J-TwxrvC881TEQ
>   * Для более точного контроля отображения вёрстки рекомендуется использовать плагин `PerfectPixel` для браузера `Chrome`. Этот инструмент позволяет накладывать изображение макета поверх вёрстки в браузере, что значительно упрощает процесс выявления и исправления визуальных несоответствий. Для этого достаточно экспортировать весь макет из Figma в формате изображения и использовать его в качестве overlay-слоя при помощи [плагина](https://chromewebstore.google.com/detail/perfectpixel-by-welldonec/dkaagdgjmgdmbnecmcefdhjekcoceebi){target="_blank"}
> 
> Можно лучше:
> 
> * Все комментарии "Можно лучше" в коде.
> * В качестве альтернативного шрифта для `PressStart2P` по типу засечек можно использовать `fantasy`
> 
> Необходимо исправить все "надо исправить". Исправление "можно лучше" остаются на ваше усмотрение.

kspshnik committed:
[Hotfix] Corrected HTML Validator issues

kspshnik committed:
[Hotfix] Removed `footer`, add button moved to `main`

kspshnik committed:
[Hotfix] Fixed animation delays on `like-icon:hover`

kspshnik committed:
[Hotfix] Corrected rest of transitions

kspshnik committed:
[Hotfix] Created `card__icon-button:focus-visible`

kspshnik committed:
[Hotfix] Changed fallback font for headings

kspshnik commented:
> Надо исправить:
> 

> * Все Error валидатора https://validator.w3.org/#validate_by_input
> * Всем кнопкам в проекте нужен атрибут `type`

Fixed.

> * Иконка лайка `like-icon` https://disk.yandex.com/i/VnAfyo-e7rU4YQ
>   
>   * Иконку лайка необходимо центрировать в кнопке(удобно использовать сетку `flex/grid`)

Fixed.

>   * Высота рамки при фокусе на иконку лайка согласно заданию сходится с высотой кнопки `Like`

Fixed.

>   * Между кнопкой с иконкой лайка и кнопкой `Like` согласно заданию есть разрыв

Fixed.

> * В качестве альтернативного шрифта для `PressStart2P` по типу засечек можно использовать `fantasy`

Fixed.

kspshnik committed:
[Hotfix/Final] Pixelperfected both checkpoints

kspshnik commented:
> Надо исправить:

> * Иконка лайка `like-icon` https://disk.yandex.com/i/VnAfyo-e7rU4YQ
>   
>   * Иконку лайка необходимо центрировать в кнопке(удобно использовать сетку `flex/grid`)
>   * Высота рамки при фокусе на иконку лайка согласно заданию сходится с высотой кнопки `Like`
>   * Между кнопкой с иконкой лайка и кнопкой `Like` согласно заданию есть разрыв
> * В проекте есть нарушения в отображении по макету

Fixed.

> 
> Необходимо исправить все "надо исправить". Исправление "можно лучше" остаются на ваше усмотрение.

All fixed.

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -21,12 +21,361 @@
+  flex-flow: column nowrap;
+  justify-content: start;
+  align-items: center;
+  ;
```
Лишняя точка с запятой

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -21,12 +21,361 @@
+{
+  opacity: 0.6;
+  pointer-events: none;
+  transition: opacity 0.3s ease;
```
Правило наследуется, нет необходимости его явно указывать

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -21,12 +21,361 @@
+.theme-menu__button:hover
+{
+  opacity: 0.6;
+  transition: opacity 0.3s ease;
```
Правило наследуется, нет необходимости его явно указывать

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -21,12 +21,361 @@
+{
+  outline: none;
+  opacity: 0.6;
+  transition: opacity 0.3s ease;
```
Правило наследуется, нет необходимости его явно указывать

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -21,12 +21,361 @@
+.theme-menu__button:focus-visible::after
+{
+  transform: scaleX(1);
+  transition: transform 0.3s ease;
```
Правило наследуется, нет необходимости его явно указывать

AliaksandrHrychonak reviewed (commented):
Добрый день, исправления отличные, осталось исправить:

- В авто тёмной теме значение вариативных шрифтов не по макету
- при наведении кнопки не должны пропадать https://disk.yandex.com/i/vriqLjaVv-MyKA , фон нужно оставлять в кнопке https://disk.yandex.com/i/PUKmFowSOBRGnQ
* Иконка лайка `like-icon` https://disk.yandex.com/i/3ZaFaQOYEf-V6g (тень не нужна)
  * Иконку лайка необходимо центрировать в кнопке(удобно использовать сетку `flex/grid`)
  * Высота рамки при фокусе на иконку лайка согласно заданию сходится с высотой кнопки `Like`
  * Между кнопкой с иконкой лайка и кнопкой `Like` согласно заданию есть разрыв

kspshnik commented:
Замечания исправлены, работа принята. Дальнейшая разработка в ветке `final`.
